installer/scripts/tomlparser.rb

```diff
@@ -121,10 +122,12 @@ def populateSettingValuesFromConfigMap(parsedConfig)
   end
 
   begin
-    if !parsedConfig.nil?  && !parsedConfig[:agent_settings][:health_model].nil? && !parsedConfig[:agent_settings][:health_model][:enabled].nil?
+    if !parsedConfig.nil? && !parsedConfig[:agent_settings].nil? && !parsedConfig[:agent_settings][:health_model].nil? && !parsedConfig[:agent_settings][:health_model][:enabled].nil?
         @enable_health_model = parsedConfig[:agent_settings][:health_model][:enabled]
-        puts "enable_health_model = #{@enable_health_model}"
+    else
+        @enable_health_model = false
```

Comment:
Given the fact you will be working on the health config for this sprint, it would make sense to move all health model to config parsing to its own code, so we can nicely keep the sub-sections (logcollection, prometheusmetriccollection, healththresholds) seperate and each idempotent. Its okay for duplicate a few lines o code as it greatly improves testability and also created less impact for customers when they err on a sub-section. Please move this to its own as part of upcoming health config work.

Reply:
I will address this as part of config PR